WordPress/Classes/Services/Domains/DomainsService+AllDomains.swift

```diff
@@ -0,0 +1,12 @@
+import Foundation
+import WordPressKit
+
+extension DomainsService {
+
+    func fetchAllDomains(params: AllDomainsEndpointParams = .init(), completion: @escaping (AllDomainsEndpointResult) -> Void) {
```

Comment:
[Suggestion] `params` being an optional parameter might lead to misuse. I suggest changing this to require explicit verbose params (resolveStatus & noWPCom). Locale can be evaluated inside the function. I also suggest adding some documentation explaining how these params affect the result.

Reply:
Good point, at first, I aimed to mimic the exact API of All Domains endpoint, and the query params were optional in the endpoint.

To clarify, are you suggesting that I should make `resolveStatus` and `noWPCOM` required instead of optional? If so, should they be assigned default values?

Reply:
I was suggesting the following
```suggestion
    func fetchAllDomains(resolveStatus: Bool, noWPCOM: Bool, completion: @escaping (AllDomainsEndpointResult) -> Void) {
        let locale = getLocale()
        let params = AllDomainsEndpointParams()
        params.resolveStatus = resolveStatus
        params.noWPCOM = noWPCOM
        params.locale = locale
        // Use params
```

Making `resolveStatus` and `noWPCOM` required instead of optional on the `AllDomainsEndpointParams` level could work too. However, this will require the dev using `fetchAllDomains` to look up `AllDomainsEndpointParams` elsewhere to understand what it does.

Reply:
Resolved in 539549cf3dbe624b1b71f1a507e4beaa06d50e9c